src/sentry/snuba/metrics/extraction.py

```diff
@@ -567,13 +568,40 @@ def combine(cls, *supported_by):
         )
 
 
+def should_use_on_demand_metrics_for_querying(organization: Organization, **kwargs) -> bool:
+    """Helper function to check if an organization can query an specific on-demand function"""
+    components = _extract_aggregate_components(kwargs["aggregate"])
+    if components is None:
+        return False
+    function, _ = components
+
+    # This helps us control which functions are allowed to use the new spec version.
+    if function in OPS_REQUIRE_FEAT_FLAG:
+        if not organization:
+            # We need to let devs writting tests that if they intend to use a function that requires a feature flag
+            # that the organization needs to be included in the test.
+            if os.environ.get("PYTEST_CURRENT_TEST"):
+                logger.error("Pass the organization to create the spec for this function.")
```

Comment:
I don't know that we like including test specific logging in production code, can we just raise an exception instead?

Reply:
@k-fish it will not log in production because of `os.environ.get("PYTEST_CURRENT_TEST")` which is only available during `pytest` runs.